agent/functional_tests/tests/generated/simpletests_unix/simpletests_generated_unix_test.go

```diff
@@ -762,6 +762,86 @@ func TestSimpleExit(t *testing.T) {
 
 }
 
+// TestTaskLocalVolume Verify that task specific Docker volume works as expected
+func TestTaskLocalVolume(t *testing.T) {
+
+	// Parallel is opt in because resource constraints could cause test failures
+	// on smaller instances
+	if os.Getenv("ECS_FUNCTIONAL_PARALLEL") != "" {
+		t.Parallel()
+	}
+	agent := RunAgent(t, nil)
+	defer agent.Cleanup()
+	agent.RequireVersion(">=1.18.0")
+
+	td, err := GetTaskDefinition("task-local-vol")
+	if err != nil {
+		t.Fatalf("Could not register task definition: %v", err)
+	}
+	testTasks, err := agent.StartMultipleTasks(t, td, 1)
+	if err != nil {
+		t.Fatalf("Could not start task: %v", err)
+	}
+	timeout, err := time.ParseDuration("2m")
+	if err != nil {
+		t.Fatalf("Could not parse timeout: %#v", err)
+	}
+
+	for _, testTask := range testTasks {
+		err = testTask.WaitStopped(timeout)
+		if err != nil {
+			t.Fatalf("Timed out waiting for task to reach stopped. Error %#v, task %#v", err, testTask)
+		}
+
+		if exit, ok := testTask.ContainerExitcode("exit"); !ok || exit != 42 {
+			t.Errorf("Expected exit to exit with 42; actually exited (%v) with %v", ok, exit)
+		}
+
+		defer agent.SweepTask(testTask)
+	}
+
+}
+
+// TestTaskSharedVolume Verify that shared Docker volume works as expected
```

Comment:
Should we add some tests here that write real data to the volume rather than just "touching" it? For example... perhaps we should have something that fills the volume to make sure it can actually be used by a task?

Reply:
A better way to test the shared volume may be actually starting two tasks:
* task one to write to the volume
* task two read from the volume and verify the content
Any thoughts?

Reply:
That's a good idea. I think we could test for volume cleanup through this too, where task 1 writes to the volume and stops executing (volume not removed here), and task 2 reads from the same volume.